aiogram/fsm/storage/mongo.py

```diff
@@ -0,0 +1,120 @@
+from typing import Any, Dict, Optional, cast
+
+from motor.motor_asyncio import AsyncIOMotorClient, AsyncIOMotorCollection
+
+from aiogram.fsm.state import State
+from aiogram.fsm.storage.base import (
+    BaseStorage,
+    DefaultKeyBuilder,
+    KeyBuilder,
+    StateType,
+    StorageKey,
+)
+
+
+class MongoStorage(BaseStorage):
+    """
+    MongoDB storage required :code:`motor` package installed (:code:`pip install motor`)
+    """
+
+    def __init__(
+        self,
+        client: AsyncIOMotorClient,
+        key_builder: Optional[KeyBuilder] = None,
+        db_name: str = "aiogram_fsm",
+        states_collection_name: str = "states",
+        data_collection_name: str = "data",
+    ) -> None:
+        """
+        :param client: Instance of AsyncIOMotorClient
+        :param key_builder: builder that helps to convert contextual key to string
+        :param db_name: name of the MongoDB database for FSM
+        :param states_collection_name: name of the collection for storing FSM states.
+        :param data_collection_name: name of the collection for storing additional data.
+        """
+        if key_builder is None:
+            key_builder = DefaultKeyBuilder()
+        self._client = client
+        self._db_name = db_name
+        self._states_collection: AsyncIOMotorCollection = self._client[db_name][
+            states_collection_name
+        ]
+        self._data_collection: AsyncIOMotorCollection = self._client[db_name][data_collection_name]
```

Comment:
Get single database instance via `self._client[db_name]` first, then reuse it for getting collection objects.
It will make code easier to read